airflow-core/src/airflow/migrations/versions/0042_3_0_0_add_uuid_primary_key_to_task_instance_.py

```diff
@@ -255,22 +255,23 @@ def upgrade():
     elif dialect_name == "sqlite":
         from uuid6 import uuid7
 
-        stmt = text("SELECT COUNT(*) FROM task_instance WHERE id IS NULL")
         conn = op.get_bind()
-        task_instances = conn.execute(stmt).scalar()
-        uuid_values = [str(uuid7()) for _ in range(task_instances)]
 
-        # Ensure `uuid_values` is a list or iterable with the UUIDs for the update.
-        stmt = text("""
-            UPDATE task_instance
-            SET id = :uuid
-            WHERE id IS NULL
-        """)
+        stmt = text("SELECT rowid FROM task_instance WHERE id IS NULL")
+        rows = conn.execute(stmt).fetchall()
 
-        for uuid_value in uuid_values:
-            conn.execute(stmt.bindparams(uuid=uuid_value))
+        update_stmt = text("UPDATE task_instance SET id = :uuid WHERE rowid = :rowid")
+
+        for row in rows:
+            conn.execute(
+                update_stmt.bindparams(
+                    uuid=str(uuid7()),
+                    rowid=row.rowid,
+                )
+            )
 
         with op.batch_alter_table("task_instance") as batch_op:
+            conn.execute(text("PRAGMA defer_foreign_keys = ON;"))
```

Comment:
Nit: We should probably do it this way, as the conn.execute isn't part of the batch_op.
```suggestion
        conn.execute(text("PRAGMA defer_foreign_keys = ON;"))
        with op.batch_alter_table("task_instance") as batch_op:
```

Reply:
Turns out this isn't needed at all, I'd mistaken which FK constraints were being violated when testing this, and it turns out it was a problem in the test data, not a problem in the migration.